UI/Dialogs/BaseDialog.cpp

```diff
@@ -1086,4 +1116,85 @@ namespace dialog
 	{
 		return m_lpMapiObjects;
 	}
+	
+	std::wstring LookupFamilyName(_In_ PCWSTR familyName, _In_ const UINT32 filter)
+	{
+		UINT32 count = 0;
+		UINT32 length = 0;
+		std::wstring buildInfo;
+		LONG rc = FindPackagesByPackageFamily(familyName, filter, &count, nullptr, &length, nullptr, nullptr);
+		if (rc == ERROR_SUCCESS)
+		{
+			output::DebugPrint(1, L"FindPackagesByPackageFamily -> No packages found\n");
+			return L"";
+		}
+		else if (rc != ERROR_INSUFFICIENT_BUFFER)
+		{
+			output::DebugPrint(1, L"Error %d in FindPackagesByPackageFamily\n");
+			return L"";
+		}
+
+		PWSTR * fullNames = (PWSTR *)malloc(count * sizeof(*fullNames));
+		if (fullNames == nullptr)
+		{
+			free(fullNames);
+			output::DebugPrint(1, L"LookupFamilyName -> Error allocating memory\n");
+			return L"";
+		}
+
+		PWSTR buffer = (PWSTR)malloc(length * sizeof(WCHAR));
+		if (buffer == nullptr)
+		{
+			free(buffer);
```

Comment:
Instead of the constant freeing, might be simpler to just go ahead and nest all these if and then free everything after.

Alternative is style like this:
```

auto buffer = something;
auto properties = something
if (buffer && properties)
{
// call FindPackagesByPackageFamily
}


if (count)
{
// loop over results
}


//do all the free
```